File: PasswordGenerator.py
import tkinter as tk
from tkinter import messagebox
import random
import string

def generate_random_character():
    characters = string.ascii_letters + string.digits + string.punctuation
    return random.choice(characters)

def generate_random_password(length):
    password = ''.join(generate_random_character() for _ in range(length))
    return password

def determine_password_strength(length):
    if length < 5:
        return "weak"
    elif 5 <= length < 10:
        return "medium"
    else:
        return "strong"

def generate_passwords():
    try:
        num_passwords = int(entry_num_passwords.get())
        password_length = int(entry_password_length.get())
        passwords = [generate_random_password(password_length) for _ in range(num_passwords)]
        strength = determine_password_strength(password_length)
        display_passwords(passwords, strength)
    except ValueError:
        messagebox.showerror("Invalid Input", "Please enter valid numbers for both fields.")

def display_passwords(passwords, strength):
    text_output.delete(1.0, tk.END)  # Clear previous output
    for i, password in enumerate(passwords, 1):
        text_output.insert(tk.END, f"Password {i}: {password}\n")
    text_output.insert(tk.END, f"\nPassword Strength: {strength}")

# Create the main window
root = tk.Tk()
root.title("Random Password Generator")

# Create and place the widgets
label_num_passwords = tk.Label(root, text="Enter the total number of random passwords:")
label_num_passwords.pack()

entry_num_passwords = tk.Entry(root)
entry_num_passwords.pack()

label_password_length = tk.Label(root, text="Enter the length of each password (At least 8 characters): ")
label_password_length.pack()

entry_password_length = tk.Entry(root)
entry_password_length.pack()

generate_button = tk.Button(root, text="Generate Passwords", command=generate_passwords)
generate_button.pack()

text_output = tk.Text(root, height=15, width=50)
text_output.pack()

# Run the application
root.mainloop()

File: PasswordGenerator.java
import java.util.Random;
import java.util.Scanner;

public class PasswordGenerator {

    private static char generateRandomCharacter() {
        String characters = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789!@#$%^&*()_+{}|:<>?-=[];,./";
        Random random = new Random();
        return characters.charAt(random.nextInt(characters.length()));
    }

    private static String generateRandomPassword(int length) {
        StringBuilder password = new StringBuilder(length);
        for (int i = 0; i < length; i++) {
            password.append(generateRandomCharacter());
        }
        return password.toString();
    }

    private static String determinePasswordStrength(String password) {
        int length = password.length();
        boolean hasLower = false, hasUpper = false, hasDigit = false, hasSpecial = false;

        for (char c : password.toCharArray()) {
            if (Character.isLowerCase(c)) hasLower = true;
            if (Character.isUpperCase(c)) hasUpper = true;
            if (Character.isDigit(c)) hasDigit = true;
            if ("!@#$%^&*()_+{}|:<>?-=[];,./".indexOf(c) >= 0) hasSpecial = true;
        }

        if (length < 5) {
            return "weak";
        } else if (length < 10) {
            return "medium";
        } else if (hasLower && hasUpper && hasDigit && hasSpecial) {
            return "strong";
        } else {
            return "medium";
        }
    }

    private static String[] generatePasswords(int numPasswords, int passwordLength) {
        String[] passwords = new String[numPasswords];
        for (int i = 0; i < numPasswords; i++) {
            passwords[i] = generateRandomPassword(passwordLength);
        }
        return passwords;
    }

    private static void printPasswords(String[] passwords) {
        for (int i = 0; i < passwords.length; i++) {
            String strength = determinePasswordStrength(passwords[i]);
            System.out.println("Password " + (i + 1) + ": " + passwords[i] + " (Strength: " + strength + ")");
        }
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        System.out.print("Enter the total number of random passwords you want: ");
        int numPasswords = scanner.nextInt();

        System.out.print("Enter the length of each password (12-15 characters is a strong password length): ");
        int passwordLength = scanner.nextInt();

        String[] passwords = generatePasswords(numPasswords, passwordLength);

        printPasswords(passwords);

        scanner.close();
    }
}
